### DISCUSSION TITLE
[Help] Register-ObjectEvent and Events that return value

### DISCUSSION THREAD

**Discussion by Hexality** [PowerShell/PowerShell#21170] [Q&A]
I have a simple question, is there any way to use Register-ObjectEvent for events that return a value without having to rely on C#? cause I don't know a damn about it and it's kinda difficulty for me to learn it...

I tried using it and it gave me this error, that's the reason of this disc.:
```ps1
Register-ObjectEvent: /mnt/MargatroidAlice/Projects/Projectt/bot.ps1:73:5
Line |
  73 |      Register-ObjectEvent -InputObject $client MessageReceived
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot register for the specified event. Events that require a return value are not supported. (Parameter 'eventName')
```

**Comment by jborean93** [comment]
Unfortunately it can't be done with `Register-ObjectEvent`. You need to define the event delegate handler in C# and use that as the subscribed delegate.
